### PR TITLE
Add experimental e2e test script

### PR DESCRIPTION
**Pull request by wongma7**
This little script should help with sanity checking the authenticator works end-to-end using the example.yaml.

It's experimental because it makes a lot of implicit assumptions (like you are currently using some IAM role) and it writes/reads/installs files all over the place.

Ideally it should be run in a container and we could set up a prow job, but I have not tested it in a container yet.

And in the future tests should be written in golang, not crappy bash. At least there should be tests for the other backend modes.

First time usage:
```
./test.sh
```

To avoid recompiling kubernetes:
```
env GO_OUT=$PWD/kubernetes/_output/bin ./test.sh
```

To skip tearing down the cluster (so you can debug)
```
env DOWN=false ./test.sh
```

### CODE REVIEW

**Comment by wongma7** [comment]
BTW, I forgot https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/313/files already existed, but anyway this can coexist with that. This uses local-up-cluster.sh and that one uses kops.

**Comment by nckturner** [comment]
This might need:  `./hack/install-etcd.sh`, when I ran it, it failed because I didn't have etcd in my path.

**Comment by wongma7** [comment]
Added  ./hack/install-etcd.sh .. tested by removing etcd from PATH and the script worked

**Comment by nckturner** [comment]
Hmm, I got: 
```
**
**Test should be authorized when authenticator is running and role is in mapRoles
**
clusterrole.rbac.authorization.k8s.io/aws-iam-authenticator created
serviceaccount/aws-iam-authenticator created
clusterrolebinding.rbac.authorization.k8s.io/aws-iam-authenticator created
configmap/aws-iam-authenticator created
daemonset.apps/aws-iam-authenticator created
waiting for daemonset ready...
waiting for daemonset ready...
daemonset ready
error: You must be logged in to the server (Unauthorized)

**
**FAIL!
**

```

File: hack/test.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env bash
+
+DOWN=${DOWN:true}
```
Comment (nckturner) [inline hack/test.sh:3]: Should this be `${DOWN:-true}`?  I am unfamiliar with just a `:`.

File: hack/test.sh [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+super_echo() {
+  echo ""
+  echo "**"
+  echo "**$1"
+  echo "**"
+}
+
+super_echo "WARNING: this script is experimental, run at your own risk!"
```
Comment (nckturner) [inline hack/test.sh:12]: I like how this prints but doesn't give you the opportunity to stop it.  Also, what is the risk?  Seems like a rather innocuous script.

File: hack/test.sh [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+pushd test-infra
+go install ./kubetest
+popd
+
+super_echo "Install hack/local-cluster-up.sh"
+if [ ! -d "kubernetes" ] ; then
+  git clone https://github.com/kubernetes/kubernetes.git
+fi
+pushd kubernetes
+git checkout release-1.17
```
Comment (nckturner) [inline hack/test.sh:27]: Can we parameterize the version?

**Comment by nckturner** [comment]
This could be causing the issue I'm getting: 

```
cp: cannot create regular file '/var/aws-iam-authenticator/cert.pem': Permission denied
cp: cannot create regular file '/var/aws-iam-authenticator/key.pem': Permission denied
cp: cannot create regular file '/etc/kubernetes/aws-iam-authenticator/kubeconfig.yaml': Permission denied

```

File: hack/test.sh (hunk 1 above)
Comment (wongma7) [reply hack/test.sh:3]: yes

File: hack/test.sh (hunk 2 above)
Comment (wongma7) [reply hack/test.sh:12]: IDK maybe I'll remove this warning. Or at least reword it to be clear that it's writing stuff into /var/kubernetes (since that's what aws-iam-authenticator init and hack/local-up-cluster.sh does)

File: hack/test.sh (hunk 3 above)
Comment (wongma7) [reply hack/test.sh:27]: workingon it
